R/explore_text_columns.R

```diff
@@ -1,19 +1,311 @@
-#' Performs EDA of text features
+#' Performs EDA of text features. Works well with Rmarkdown.
+#' Its highly recommended to use code-chunk option results='asis'
+#' to make use of the markdown syntax
+#' - identifies text columns if not already specified
 #' - prints the summary statistics of character length
 #' - plots the distribution of character length
 #' - prints the summary statistics of word count
 #' - plots the distribution of word count
 #' - plots the word cloud
+#' - plots a bar plot of top 10 words
+#' - plot a word cloud of bigrams
+#' - plot a bar plot of top 10 bigrams
+#'
+#' @importFrom dplyr %>%
 #'
 #' @param df dataframe: dataset
-#' @param text_cols vector of numeric column names
+#' @param text_cols vector of text column names
 #'
-#' @return a list of plots
+#' @return a list of results and plots
+#' @export
 #'
 #' @examples
-#' \dontrun{
-#' explore_text_columns(df)
-#' }
-explore_text_columns <- function(df, text_cols=NULL){
+#' explore_text_columns(cars)
+explore_text_columns <- function(df, text_cols=vector(mode='character')) {
+
+  results <- list()
+  word_count <- frequen <- NULL
+
+  if (!is.character(text_cols)) {
+    stop("text_cols is not passed as a character vector")
+  }
+
+  if (!is.data.frame(df)) {
+    stop("The df passed is not a Data Frame")
+  }
+
+  if(length(text_cols) == 0) {
+    for (col in colnames(df %>% dplyr::select_if(is.character))){
```

Comment:
I would suggest adding some comments here on how you retrieve the text columns.

Reply:
I've added comments to the entire code now
I've addressed this is this [commit](https://github.com/UBC-MDS/datascience.eda.R/pull/15/commits/84f285d351b363b8399d26529026ad0be8e11c7a)